crates/ide_completion/src/item.rs

```diff
@@ -292,6 +294,81 @@ impl CompletionItem {
         }
     }
 
+    /// Used when creating a completion item that includes type information.
+    ///
+    /// Relevance and ref_match are pre-populated based on the provided type
+    /// information.
+    pub(crate) fn new_with_type_info(
+        ctx: &CompletionContext,
+        completion_kind: CompletionKind,
+        source_range: TextRange,
+        label: impl Into<String>,
```

Comment:
I am not sure that just using label is the right call -- label is, fundamentally, a UI concern, and it doesn't need to correspond to the actual semantic name. For macros I think we add `!` to label, and it'd be weird if that stylistic choice affected relevance.

Reply:
I've renamed this parameter to `completion_name` in the new helper method.